iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepGrammar.java

```diff
@@ -84,20 +87,36 @@ public Expression EXPRESSION() {
           STRING_LITERAL_VALUE())));
   }
 
-  // TODO SONARIAC-934 Extending of LiteralValue in grammar should return the proper Expression implementation
-  public StringLiteral LITERAL_VALUE() {
-    return b.<StringLiteral>nonterminal(BicepLexicalGrammar.LITERAL_VALUE).is(
-      f.stringLiteral(
+  public Expression LITERAL_VALUE() {
+    return b.<Expression>nonterminal(BicepLexicalGrammar.LITERAL_VALUE).is(
+      b.firstOf(
+        NUMERIC_LITERAL(),
+        BOOLEAN_LITERAL(),
+        NULL_LITERAL()));
+  }
+
+  public NumericLiteral NUMERIC_LITERAL() {
+    return b.<NumericLiteral>nonterminal(BicepLexicalGrammar.NUMERIC_LITERAL).is(
+      f.numericLiteral(b.token(BicepLexicalGrammar.NUMERIC_LITERAL_VALUE)));
+  }
+
+  public BooleanLiteral BOOLEAN_LITERAL() {
+    return b.<BooleanLiteral>nonterminal(BicepLexicalGrammar.BOOLEAN_LITERAL).is(
+      f.booleanLiteral(
         b.firstOf(
-          b.token(BicepLexicalGrammar.NUMBER_LITERAL),
-          b.token(BicepLexicalGrammar.TRUE_LITERAL),
-          b.token(BicepLexicalGrammar.FALSE_LITERAL),
-          b.token(BicepLexicalGrammar.NULL_LITERAL))));
+          b.token(BicepLexicalGrammar.TRUE_LITERAL_VALUE),
+          b.token(BicepLexicalGrammar.FALSE_LITERAL_VALUE))));
+  }
+
+  public NullLiteral NULL_LITERAL() {
+    return b.<NullLiteral>nonterminal(BicepLexicalGrammar.NULL_LITERAL).is(
+      f.nullLiteral(
+        b.token(BicepLexicalGrammar.NULL_LITERAL_VALUE)));
   }
 
   public StringLiteral STRING_LITERAL_VALUE() {
```

Comment:
Should it be `STRING_LITERAL` or `STRING_LITERAL_VALUE`?

On a side note, I feel like naming for strings is a bit confusing, as the grammar doesn't have `stringLiteral` and `stringValue`, but rather has `stringComplete` as a variant of `interpString` without any interpolation. In this sense our `STRING_LITERAL_VALUE` is `stringComplete`

Reply:
Good point, I renamed it to `STRING_LITERAL`.
I agree, the name is not great and is not reflected in the grammar. It could be interpreted as an equivalent of `stringComplete`, but as mentioned earlier this is more like a temp approach to support string, that will be removed later in favor of a more complete version aligned to the Bicep grammar.